src/cpu/o3/rename.cc

```diff
@@ -286,7 +286,7 @@ Rename::setActiveThreads(std::list<ThreadID> *at_ptr)
 
 
 void
-Rename::setRenameMap(UnifiedRenameMap rm_ptr[])
+Rename::setRenameMap(UnifiedRenameMap rm_ptr[MaxThreads])
```

Comment:
It might be cleaner to just introduce a `PerThreadUnifiedRenameMap` (or something along those lines) that is a std::array<UnifiedRenameMap, MaxThreads>. That would make it behave as a normal C++ container but with the performance of a plain array. I realise this is a slightly bigger change.

Reply:
Right now, we want to prioritize getting this change in so we can test the new compilers in our weekly tests, but I introduced this in an issue (#373 ).  I don't have the time to make this change right now, but this way it can still be addressed.